app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,79 @@
+class ProjectsController < ApplicationController
+  before_action :set_project, only: [:show, :edit, :update, :destroy]
+  before_action :validates_role, only: [:new, :edit, :update, :create]
+  before_action :validates_admin, only: :destroy
+
+  def index
+    @projects = Project.all
+  end
+
+  def show
+  end
+
+  def new
+    @project = Project.new
+    get_clients
+    get_users
+  end
+
+  def edit
+    get_clients
+    get_users
+  end
+
+  def create
+    @project = Project.new(project_params)
+    params[:users][:id].each do |user|
```

Comment:
:(
Discuss this with Ammad first thing in the morning.
Do see how has many through associations are saved.

Reply:
always look for something that resembles what you are trying to do.

Reply:
ask if you are still confused.